iOS Client/PodcastsController/PodcastsCoordinator.swift

```diff
@@ -81,4 +104,27 @@ class PodcastsCoordinator: Coordinator, StoreSubscriber {
             player.play()
         }
     }
+    
+    func getCurrentDuration() -> Double {
+        let time = player.currentTime()
+        return time.seconds
+    }
+    
+    // MARK: - AVAudioPlayerDelegate
+    func audioPlayerDidFinishPlaying(_ player: AVAudioPlayer, successfully flag: Bool) {
+        if flag {
+            MBStore.sharedStore.dispatch(FinishedPodcast())
+        } else {
+            MBStore.sharedStore.dispatch(PodcastError())
+        }
+    }
+    
+    func seek(to second: Double) {
+        let time = CMTime(seconds: second, preferredTimescale: 1)
+        player.seek(to: time)
+    }
+}
+
+protocol PodcastHandler {
```

Comment:
`PodcastHandler` and `PodcastPlayerDelegate` protocol definitions should be swapped in terms of placement.

Think about how table view works. Right next to table view, they define `TableViewDataSource` and `TableViewDelegate` protocols. And the table view has properties of those types. Then outsiders can conform to those protocols and interface with table view.

In this case, the coordinator has a property of type `PodcastPlayerDelegate`. That protocol should be defined in this file. Then, the Podcast View Controller could implement it. Similarly, the Podcast View controller file should hold the definition of `PodcastHandler`. 

However, again, all of this is a moot point b/c we shouldn't have two-way delegation going on. This coordinator should be calling functions directly on the view controller. We don't need protocols for that direction. However, if you want to setup a callback scheme from the view controller back up to the coordinator, then delegation is the move for that direction. You'd define some delegate protocol in the view controller file and then conform to it in the coordinator file and directly set the view controller's delegate to self (the coordinator).

Think of delegation as a solution to when you need a callback. 

Remember, we know all about the table view, so we can tell it to do whatever we want without delegation. But the table view has no idea who we are, so we set its delegate property in order to receive callbacks. All the table view knows is that we conform to its delegate protocol.

With these thoughts in mind, I think this code can be made much simpler, which I thought was done earlier on b/c I'm pretty sure we had this discussion, but maybe GitHub is being weird.

Reply:
Yeah man so I just never merged those changes back into this branch, but it should be good to go now if you want to give it a rereview. Not sure which of these comments are still relevant so I would say give it a good look over the whole thing